databricks_cli/pipelines/cli.py

```diff
@@ -95,14 +95,19 @@ def deploy_cli(api_client, spec_arg, spec, allow_duplicate_names, no_update_spec
         click.echo("Successfully created pipeline: {}".format(
             _get_pipeline_url(api_client, new_pipeline_id)))
 
-        if not no_update_spec:
-            spec_obj['id'] = new_pipeline_id
-            _write_spec(src, spec_obj)
-            click.echo("Updated spec at {} with ID {}".format(src, new_pipeline_id))
-        else:
-            click.echo("Pipeline has been assigned ID {}".format(new_pipeline_id))
+        click.echo("Pipeline has been assigned ID {}".format(new_pipeline_id))
+        click.echo(new_pipeline_id, err=True)
     else:
+        if (pipeline_id and 'id' in spec_obj) and pipeline_id != spec_obj["id"]:
+            raise ValueError(
+                "The ID provided in --pipeline_id '{}' is different from the id provided the "
+                "spec '{}'. Please resolve the conflict and try the command again."
+                .format(pipeline_id, spec["id"])
+            )
```

Comment:
I'd say something like "Because pipeline IDs are no longer persisted after being deleted, we recommend removing the ID field from your spec."

Reply:
Done.

Reply:
Sorry, meant to add that in addition to what you had before. It's still good to tell the user they've passed conflicting id params.

Reply:
Ah, understood. Updated.